doc/tutorials/your_first_project/your_first_project.rst

```diff
@@ -0,0 +1,257 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This first block of code is a bit of boilerplate but you should be used to seeing this from the ROS 2 tutorials.
+
+.. code-block:: C++
+
+  #include <memory>
+
+  #include <rclcpp/rclcpp.hpp>
+  #include <moveit/move_group_interface/move_group_interface.h>
+
+  int main(int argc, char * argv[])
+  {
+    // Initialize ROS and create the Node
+    rclcpp::init(argc, argv);
+    auto const node = std::make_shared<rclcpp::Node>(
+      "hello_moveit",
+      rclcpp::NodeOptions().automatically_declare_parameters_from_overrides(true)
+    );
+
+    // Create a ROS logger
+    auto const logger = rclcpp::get_logger("hello_moveit");
+
+    // Next step goes here
+
+    // Shutdown ROS
+    rclcpp::shutdown();
+    return 0;
+  }
+
+2.1 Build and Run
+~~~~~~~~~~~~~~~~~
+
+We will build and run the program to see that everything is right before we move on.
+
+Change directory back to the workspace directory ``ws_moveit2`` and run this command:
+
+.. code-block:: bash
+
+  colcon build --mixin debug
+
+After this succeeds, **open a new terminal**, then source the workspace environment script in that new terminal so that we can run our program.
+
+.. code-block:: bash
+
+  cd ~/ws_moveit2
+  source install/setup.bash
+
+Run your program and see the output.
+
+.. code-block:: bash
+
+  ros2 run hello_moveit hello_moveit
+
+The program should run and exit without error.
+
+2.2 Examine the code
+~~~~~~~~~~~~~~~~~~~~
+
+The headers included at the top are just some standard C++ headers and the header for ROS and MoveIt that we will use later.
+
+After that we have the normal call to initialize rclcpp and then we create our Node.
+
+.. code-block:: C++
+
+  auto const node = std::make_shared<rclcpp::Node>(
+    "hello_moveit",
+    rclcpp::NodeOptions().automatically_declare_parameters_from_overrides(true)
+  );
+
+The first argument is a string that ROS will use to make a unique node.
+The second is needed for MoveIt because of how we use ROS Parameters.
+
+Lastly we have the code to shutdown ROS.
+
+3 Plan and Execute using MoveGroupInterface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In place of the comment that says "Next step goes here" add this code:
+
+.. code-block:: C++
+
+  // Create the MoveIt MoveGroup Interface
+  using moveit::planning_interface::MoveGroupInterface;
+  auto move_group_interface = MoveGroupInterface(node, "panda_arm");
+
+  // Set a target Pose
+  auto const target_pose = []{
+    geometry_msgs::msg::Pose msg;
+    msg.orientation.w = 1.0;
+    msg.position.x = 0.28;
+    msg.position.y = -0.2;
+    msg.position.z = 0.5;
+    return msg;
+  }();
+  move_group_interface.setPoseTarget(target_pose);
+
+  // Create a plan to that target pose
+  auto const [success, plan] = [&move_group_interface]{
+    moveit::planning_interface::MoveGroupInterface::Plan msg;
+    auto const ok = static_cast<bool>(move_group_interface.plan(msg));
+    return std::make_pair(ok, msg);
+  }();
```

Comment:
I think it might be helpful to add a comment directing to your note about lambdas below. I was confused about it until I got to the end of the tutorial.

Reply:
I have updated this based on your feedback, would you mind looking over my changes and letting me know if you think they are what you were hoping for?

Reply:
lgtm!